### PR TITLE
Improve iterator performance

### PR DESCRIPTION
**Pull request by luis-pereira**

### CODE REVIEW

File: lxqtsettings.cpp [hunk 1]
```diff
@@ -631,10 +631,12 @@ SettingsCache::SettingsCache(QSettings *settings) :
...
-   foreach (QString key, mSettings.allKeys())
-   {
-       mCache.insert(key, mSettings.value(key));
-   }
+    const QStringList keys = mSettings.allKeys();
+
+    const int N = keys.size();
+    for (int i = 0; i < N; ++i) {
+        mCache.insert(keys.at(i), mSettings.value(keys.at(i)));
+    }
```
Comment (paulolieuthier) [inline lxqtsettings.cpp]: How is this faster than the regular range-based for with const reference, which is more readable?

Comment (luis-pereira) [inline lxqtsettings.cpp]: It automatically takes a copy of the container when it enters a foreach loop.

Comment (paulolieuthier) [inline lxqtsettings.cpp]: I see, you're right. Seeing the implementation of `allKeys()`, I say it's quite expensive.

**Comment by paulolieuthier** [comment]
Ship it.

**Pull request merged**